web/modules/custom/custom_page/custom_page.module

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @file
+ * Main functions and hook implementations of the custom page module.
+ */
+
+use \Drupal\og\Entity\OgMembership;
+use \Drupal\user\Entity\User;
+use \Drupal\Core\Cache\Cache;
+
+/**
+ * Implements hook_og_membership_insert().
+ *
+ * We will invalidate the specific user's rdf_entity tag based cache
+ * every time a new membership is inserted..
+ * The reason behind this is that we want to cache content per users
+ * e.g. blocks. OG invalidates the cache of the specific user's membership
+ * when the membership is updated or deleted but flushes all membership list
+ * when a new membership is created.
+ * This will have an impact to the performance if we had to flush all the
+ * membership specific content for all users every time a membership is created.
+ *
+ * @var \Drupal\og\Entity\OgMembership $og_membership
+ *  The membership that has just been inserted.
+ */
+function custom_page_og_membership_insert(OgMembership $og_membership) {
+  $account = \Drupal::currentUser();
+  $current_user = User::load($account->id());
+  $tag = 'rdf_entity:' . $current_user->getEntityTypeId() . ':' . $current_user->id();
+
+  Cache::invalidateTags([$tag]);
```

Comment:
This is not reliable. You cannot expect that the currently logged in user is the one whose membership is being inserted. Administrators will also be able to let users join or leave their groups, and we can do this by direct API calls (e.g. through drush commands, cron jobs, import tasks, ...) in which case no user will be logged in.

In all these cases the wrong cache will be flushed, and the caches that are actually affected will not be flushed.

It also seems that this is flushing way too much, it flushes the entire cache tag of the user. We only need to flush the caches of the elements that are actually depending on the membership.